pkg/render/logstorage.go

```diff
@@ -359,14 +360,9 @@ func (es *elasticsearchComponent) podTemplate() corev1.PodTemplateSpec {
 	}
 
 	sc := securitycontext.NewRootContext(false)
-	// These capabilities are required for docker-entrypoint.sh.
-	// See: https://github.com/elastic/elasticsearch/blob/7.17/distribution/docker/src/docker/bin/docker-entrypoint.sh.
-	// TODO Consider removing for Elasticsearch v8+.
-	sc.Capabilities.Add = []corev1.Capability{
-		"SETGID",
-		"SETUID",
-		"SYS_CHROOT",
-	}
+	// Set the user and group to be the default elasticsearch ID
+	sc.RunAsUser = ptr.Int64ToPtr(1000)
```

Comment:
IIUC, the init containers still need to run as root to:
- set os settings
- chown the file system
and so the namespace still needs to have the privileged context. The main container now runs as 1000:1000, which is the es user. This is also the user the prepareFS script will chown to.

Is my understanding correct? Have we considered changing the user to 10000:10000 (our default user)?

Reply:
I believe this SecurityContext is for the main ElasticSearch container. The SecurityContext for each init container still uses the root context.

Reply:
Init container "elastic-internal-init-filesystem" still run as root because we need to set the vm max map count "echo 262144 > /proc/sys/vm/max_map_count". This change is only for elasticsearch container.
Currently, Elasticsearch 7, we override the security context of the Elasticsearch container to run as the root user and group. The [docker-entrypoint.sh](https://github.com/elastic/elasticsearch/blob/7.17/distribution/docker/src/docker/bin/docker-entrypoint.sh) script in Elasticsearch contained logic to switch to user 1000 when it encountered a root user. However, this logic has been removed in Elasticsearch 8. Since the ownership of the /data folder is set to the default Elasticsearch UID, we are encountering an unwritable error when running as the root user.

Reply:
We spoke offline about a potential follow-up to explore using uid:gid of 10001:10001, but continue with the PR in its current form.